Feat/entity/currency

essteer opened the pull request:
Adds a `Currency` class with two attributes:

- Primary key of `currencyCode` - validation in place to ensure it is a three-letter String
- `currencyName` - validation in place to ensure it is not null

Unit tests added and passing.

essteer committed:
feat: Add Currency class and passing unit tests

essteer committed:
feat: Add repo, service and controller layers for Currency entity, with customer RecordNotFoundException and passing unit tests for CurrencyService

dexterklui commented on src/main/java/com/fdmgroup/forex/services/CurrencyService.java:
```diff
@@ -0,0 +1,33 @@
+            new RecordNotFoundException("Currency with id '" + id + "'' not found"));
+    }
+
+    public Currency findByCurrencyName(String currencyName) throws RecordNotFoundException {
```
I don't think we will ever need this service?

dexterklui commented on src/main/java/com/fdmgroup/forex/services/CurrencyService.java:
```diff
@@ -0,0 +1,33 @@
+
+@Service
+public class CurrencyService {
+
```
I think what is needed the most is `findAllCurrencies` method?
I imagine frontend will need to fetch all currencies right in the beginning and store it for later use:
e.g. when displaying each portfolio asset, frontend will need to use its stored currencies info and show the current currency name.

essteer committed:
refactor: Remove findByCurrencyName method

essteer committed:
feat: Add findAllCurrencies method and remove unit tests for removed findCurrenciesByName method

essteer committed:
feat: Add mappings to get all currencies and get currency by id

essteer commented:
@dexterklui Noted, thanks - I've removed the findByCurrencyName method and added findAllCurrencies. Mappings have been added to the controller to get all currencies and get a single currency by ID.

essteer committed:
test: Add unit test for findAllCurrencies method

essteer committed:
feat: Add @Column annotation to Currency entity with unique constraint

essteer committed:
refactor: Amend Currency classes in line with agreed conventions

essteer committed:
feat: Add ExchangeRate entity and passing unit tests

essteer committed:
feat: Add repo and service layer with passing tests for ExchangeRate

essteer committed:
feat: Add controller layer for ExchangeRate

essteer committed:
Merge pull request #4 from forex-agile/feat/entity/exchangeRates

Feat/entity/exchange rates

essteer committed:
refactor: Amend Currency and ExchangeRate variable and method names in line with agreed conventions

essteer committed:
refactor: Rename ExchangeRate and related classes methods and variables to FxRate

dexterklui commented on src/main/java/com/fdmgroup/forex/models/Currency.java:
```diff
@@ -0,0 +1,47 @@
+    private String currencyCode;
+
+    @Column(unique = true, nullable = false)
+    private String currencyName;
```
Add `@Pattern` validation

essteer replied:
Resolved in 6ab42a9

essteer committed:
feat: Add @Pattern annotation to  field to enforce 3-letter string format

dexterklui commented on src/main/java/com/fdmgroup/forex/models/Currency.java:
```diff
@@ -0,0 +1,47 @@
+        }
+    }
+
+    public String toString() {
```
Add `@Override`

essteer replied:
Resolved in 1b92d44

essteer committed:
refactor: Add  annotation to  method
